src/util/psbt/raw.rs

```diff
@@ -154,7 +155,13 @@ impl<Subtype> Decodable for ProprietaryKey<Subtype> where Subtype: Copy + From<u
         let prefix = Vec::<u8>::consensus_decode(&mut d)?;
         let mut key = vec![];
         let subtype = Subtype::from(d.read_u8()?);
-        d.read_to_end(&mut key)?;
+        // core2 doesn't have read_to_end
+        let mut buf = [0u8; 64];
+        loop {
+            let n = d.read(&mut buf).unwrap();
```

Comment:
This is most likely a DoS vector unless there is an argument why `read` can't fail. If so I'd appreciate it being documented.

Reply:
My mistake, it should be a `?` operator

Reply:
I don't think just a ? Is sufficient https://doc.rust-lang.org/src/std/io/mod.rs.html#392

Reply:
TIL: there exist recoverable IO errors, namely `ErrorKind::Interrupted`. At lest I think that's what you mean.

Reply:
Ah, good point @TheBlueMatt.  Done